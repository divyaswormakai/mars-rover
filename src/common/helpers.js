import React from 'react';

import { KEYBOARDS_CODES, ROVER_DIRECTION, ROVER_MOVEMENT } from './constants';

const isCoordinateInsideGrid = (grid, newPosition) =>
  newPosition.x >= 0 &&
  newPosition.x <= grid.x &&
  newPosition.y >= 0 &&
  newPosition.y <= grid.y;

export const getDirectionArrow = direction => {
  switch (direction) {
    case ROVER_DIRECTION.W:
      return <>&#8592;</>;

    case ROVER_DIRECTION.E:
      return <>&#8594;</>;

    case ROVER_DIRECTION.S:
      return <>&#8595;</>;

    case ROVER_DIRECTION.N:
    default:
      return <>&#8593;</>;
  }
};

export const getRandomCoordinates = ({ x, y }) => {
  const coordinates = [];

    const newX = Math.floor(Math.random() * (x - 0));
    const newY = Math.floor(Math.random() * (y - 0));
   
      coordinates.push({
        x: newX,
        y: newY,
      });
    

  return [...new Set(coordinates)];
};

export const getRandomRoverPosition = (grid) => {
  const roverPosition = getRandomCoordinates(grid)[0];

  return roverPosition
  

};

export const getRoverMovementFromCode = code => {
  switch (code) {
    case KEYBOARDS_CODES.A:
      return ROVER_MOVEMENT.L;
    case KEYBOARDS_CODES.D:
      return ROVER_MOVEMENT.R;
    case KEYBOARDS_CODES.W:
      return ROVER_MOVEMENT.F;
    default:
      return null;
  }
};

export const getValidInstructions = instruction =>
  instruction
    ?.toUpperCase?.()
    .split('')
    .filter(item => Object.keys(ROVER_MOVEMENT).includes(item))
    .join('') ?? '';

export const isCorrectMovement = ({
  newPosition,
  grid,
}) => {
  const isInsideGrid = isCoordinateInsideGrid(grid, newPosition);
  return  isInsideGrid;
};
