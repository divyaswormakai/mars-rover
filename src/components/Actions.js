import React from 'react';
import styled from 'styled-components';

import InstructionsForm from './InstructionsForm';

const Actions = ({ className }) => (
  <div className={className}>
    <InstructionsForm />
  </div>
);

const styledActions = styled(Actions)`
  display: inline-block;
  overflow: hidden;
  border: 1px solid black;
  padding: 20px;
  width: 200px;
  border-radius: 10px;

  @media (max-width: 900px) {
    margin-top: 15px;
  }
`;

export default styledActions;
