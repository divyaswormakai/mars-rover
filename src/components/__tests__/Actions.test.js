import 'jest-styled-components';

import { render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';

import { mockComponent } from '../../common/testHelpers';
import Actions from '../Actions';

jest.mock('../InstructionsForm', () => props =>
  mockComponent('InstructionsForm', props)
);


describe('Actions test suite', () => {
  const component = <Actions />;

  const setup = () => render(component);

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
