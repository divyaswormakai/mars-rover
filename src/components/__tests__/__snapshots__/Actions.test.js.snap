// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions test suite Should match snapshot 1`] = `
.c0 {
  display: inline-block;
  overflow: hidden;
  border: 1px solid black;
  padding: 20px;
  width: 200px;
  border-radius: 10px;
}

@media (max-width:900px) {
  .c0 {
    margin-top: 15px;
  }
}

<div>
  <div
    class="c0"
  >
    <code
      aria-label="InstructionsForm"
      name="InstructionsForm"
      props="[object Object]"
    />
  </div>
</div>
`;
